ui/gtk/main_window.py

```diff
@@ -63,12 +82,134 @@ def __init__(self, locale_general: LanguageManagement, config_general: dict, con
+            label_in_box = TitleLabel(current_page.get_page_id(), current_page.get_page_sidebar_title())
+            if label_in_box.get_page_id() == self._pages[self._page_index].get_page_id():
+                label_in_box.get_style_context().remove_class('dim-label')
+            else:
+                label_in_box.get_style_context().add_class('dim-label')
+            self._component.get_component('main_window_left_list_text_box').pack_start(label_in_box, False,
+                                                                                       False, 0)
+            current_page.refresh_ui_language()
+        self._component.get_component('main_window_left_list_text_box').show_all()
+
+        self.set_title(self._pages[self._page_index].get_page_title())
+        self._component.get_component('main_window_prev_btn').set_label(
+            self._locale_general.translate_msg('main_window', 'bot_right_prev_btn'))
+        self._component.get_component('main_window_next_btn').set_label(
+            self._locale_general.translate_msg('main_window', 'bot_right_next_btn'))
+        self._component.get_component('main_window_header_right_quit_btn').set_label(
+            self._locale_general.translate_msg('main_window', 'top_right_quit_btn'))
+
+    def next_page_btn(self) -> None:
+        idx = self._page_index + 1
+        if idx >= len(self._pages):
+            return
+        self._page_index = idx
+        self.update_page()
+
+    def prev_page_btn(self) -> None:
+        idx = self._page_index - 1
+        if idx < 0:
+            return
+        self._page_index = idx
+        self.update_page()
+
+    def update_page(self) -> None:
+        self.set_title(self._pages[self._page_index].get_page_title())
+
+        if self._page_index == len(self._pages) - 1:
+            self.set_can_next(False)
+            self.set_hide_next(True)
+        else:
+            self.set_can_next(True)
+            self.set_hide_next(False)
+
+        if self._page_index == 0:
+            self.set_can_prev(False)
+            self.set_hide_prev(True)
+        else:
+            self.set_can_prev(True)
+            self.set_hide_prev(False)
+
+        self._pages[self._page_index].load_page()
+
+        pixbuf = GdkPixbuf.Pixbuf.new_from_file_at_scale(self._pages[self._page_index].get_page_icon(), 50, 50, False)
+        self._component.get_component('main_window_top_left_logo').set_from_pixbuf(pixbuf)
+        for label in self._component.get_component('main_window_left_list_text_box').get_children():
+            if label.get_page_id() == self._pages[self._page_index].get_page_id():
+                label.get_style_context().remove_class('dim-label')
+            else:
+                label.get_style_context().add_class('dim-label')
+        self._component.get_component('main_window_right_stack').set_visible_child_name(
+            self._pages[self._page_index].get_page_id())
+
+    def set_can_prev(self, can_prev: bool) -> None:
```

Comment:
I'm not a fan of the `set_[verb]_[object]` pattern. It creates a million of function for all actions of all objects on the window.

An idea to help you out:
Create an enum `MainWindowButton` with all the buttons, for example `NEXT='main_window_next_btn'`, etc. and create functions like `set_button_visibility(button: MainWindowButton, status: Bool)`
Do not add the condition `if self.page_index` as you're already doing that condition when calling the function

Note that the term `set_hide_xxx` while taking a parameter that states if we want to hide or show the element seems conflictual: the parameters don't match the function name. Prefer neutral names like 'set_visibility', etc.